### PR TITLE
HTTP-74 Add path parameter support

### LINKED ISSUE #74: Support for API path parameters
Among the different types of API parameters, it appears the path parameters (for instance `/customers/{customerId}/orders/{orderId}`) are currently not supported. 
At least it looks so by looking at the code, tests, and documentation.

The URL can be augmented to provide the path segments as a const/literals statically, at connector configuration time. But, in most cases, the path parameters need to be fed from table columns.

Path parameters being by definition required (not optional), the current limitation makes it impossible to use the connector for many APIs.

A solution would need to allow the coexistence of 
* query and path parameters for GET, and 
* query, path, and request body parameters for POST.

### PR DESCRIPTION
**Pull request by davidradl**
#### Description

add path patameter support

Resolves #74 

##### PR Checklist
- [x] Tests added
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

**Comment by davidradl** [comment]
@OlivierZembri please could you review and see if this would meet your needs?

File: src/main/java/com/getindata/connectors/http/internal/table/lookup/GetRequestFactory.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ URI constructGetUri(LookupQueryInfo lookupQueryInfo) {
             resolvedUrl.append(baseUrl.contains("?") ? "&" : "?")
                        .append(lookupQueryInfo.getLookupQuery());
         }
-
+        resolvedUrl = addPathParameters(lookupQueryInfo, resolvedUrl);
```
[the remark this answers is not in the document]
Comment (davidradl) [reply src/main/java/com/getindata/connectors/http/internal/table/lookup/GetRequestFactory.java:63]: @OlivierZembri yes that is better. Thanks

**Review by AdrianVasiliu — COMMENTED** [review]
Nicely does the trick, thanks David.

I think it would be worth adding an entry for it under https://github.com/getindata/flink-http-connector/blob/c2f3e7bdc6b9d03bf3b807ac30a1302f026f82dc/CHANGELOG.md#unreleased.

**Review by AdrianVasiliu — APPROVED** [review]
Thanks for the addition to the changelog. From my chair, approving.

**Review by OlivierZembri — APPROVED** [review]

File: src/test/java/com/getindata/connectors/http/internal/table/lookup/LookupQueryInfoTest.java [hunk 2]
```diff
@@ -35,10 +35,54 @@ public void testEmptyLookupQueryInfo() {
...
-        lookupQueryInfo = new LookupQueryInfo(null, bodyBasedUrlQueryParameters);
+        lookupQueryInfo = new LookupQueryInfo(null, bodyBasedUrlQueryParameters, null);
 
         assertThat(lookupQueryInfo.hasLookupQuery()).isFalse();
         assertThat(lookupQueryInfo.hasBodyBasedUrlQueryParameters()).isTrue();
         assertThat(lookupQueryInfo.getBodyBasedUrlQueryParameters()).isEqualTo("key1=value1");
     }
+
+    @Test
+    public void testPathBasedUrlParams1() {
```
Comment (grzegorz8) [inline src/test/java/com/getindata/connectors/http/internal/table/lookup/LookupQueryInfoTest.java:46]: nit: Could you provide more meaningful test names? So the difference between `testPathBasedUrlParams1` and `testPathBasedUrlParams2` is clear just by reading test name.

File: src/main/java/com/getindata/connectors/http/internal/table/lookup/LookupQueryInfo.java [hunk 3]
```diff
@@ -26,32 +27,45 @@ public class LookupQueryInfo implements Serializable {
...
+    }
+
     public boolean hasLookupQuery() {
         return !lookupQuery.isBlank();
     }
     public boolean hasBodyBasedUrlQueryParameters() {
         return !bodyBasedUrlQueryParams.isEmpty();
     }
-
+    public boolean hasPathBasedUrlParameters() {
```
Comment (grzegorz8) [inline src/main/java/com/getindata/connectors/http/internal/table/lookup/LookupQueryInfo.java:68]: nit: please add empty line between methods.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,9 +2,17 @@
...
+-   Changed  [LookupQueryInfo](src/main/java/com/getindata/connectors/http/internal/table/lookup/LookupQueryInfo.java)
+    Any custom implementation of this interface that aims to provide path-based requests is able to provide
+    the lookup query url with parameters surrounded by curly brackets. For example the supplied
+    URL `http://service/{customerId}`, will result in the lookup parameter `customerId` value being used
+    in the url.
+
 ### Fixed
-   
-    Moved junit support to junit 5, allowing junits to be run against flink 1.17 and 1.18.
+
```
Comment (grzegorz8) [inline CHANGELOG.md:14]: nit: remove whitespaces in line 14.

**Review by grzegorz8 — APPROVED** [review]

**Comment by davidradl** [comment]
@grzegorz8 I am not sure what happened. it says I closed this pr , while trying to make the changes you suggested. I cannot see the change in main so it does not seem to be merged. I do not have the authority to reopen this pr.

**Pull request merged**